File: Version 1.0/soaint-sgd-ui/soaint-sgd-web-api-gateway/src/main/java/co/com/foundation/sgd/dto/TareaDTO.java
package co.com.foundation.sgd.dto;

import lombok.*;

import javax.xml.bind.annotation.XmlRootElement;
import java.util.Date;
import java.util.Map;

@Data()
@ToString
@XmlRootElement
@AllArgsConstructor
@NoArgsConstructor
@Builder(builderMethodName = "newBuilder")
public class TareaDTO {

    private Long idTarea;
    private String nombre;
    private String estado;
    private Integer prioridad;
    private String idResponsable;
    private String idCreador;
    private Date fechaCreada;
    private Date tiempoActivacion;
    private Date tiempoExpiracion;
    private String idProceso;
    private Long idInstanciaProceso;
    private String idDespliegue;
    private Long idParent;
    private String descripcion;
    private String codigoDependencia;
    private String rol;
    private String nombreProceso;
    private Map<String, Object> variables;

}

File: Version 1.0/soaint-sgd-ui/soaint-sgd-web/src/app/domain/tareaDTO.ts
type tareaStatus = 'RESERVADO' | 'ENPROGRESO' | 'LISTO' | 'CANCELADO' | 'COMPLETADO';

export interface TareaDTO {
  idTarea: string;
  nombre: string;
  estado: tareaStatus;
  prioridad: number;
  idResponsable: string;
  idCreador: string;
  fechaCreada: Date;
  tiempoActivacion: Date;
  tiempoExpiracion: Date;
  idProceso: string;
  idInstanciaProceso: string;
  idDespliegue: string;
  idParent: number;
  variables: any;
  rol?:string;
  descripcion?:string;
  nombreProceso?:string;
}
